zerver/lib/email_notifications.py

```diff
@@ -117,6 +118,54 @@ def make_emoji_img_elem(emoji_span_elem: CSSSelector) -> Dict[str, Any]:
     content = lxml.html.tostring(fragment).decode('utf-8')
     return content
 
+def fix_spoilers_in_html(content: str, language: str) -> str:
+    with override_language(language):
+        spoiler_title: str = _("Open Zulip to see the spoiler content")
+    fragment = lxml.html.fromstring(content)
+    spoilers = fragment.find_class("spoiler-block")
+    for spoiler in spoilers:
+        header = spoiler.find_class("spoiler-header")[0]
+        spoiler_content = spoiler.find_class("spoiler-content")[0]
+        header_content = header.find("p")
+        if header_content is None:
+            # Create a new element to append the spoiler to)
+            header_content = lxml.html.fromstring("<p></p>")
+            header.append(header_content)
+        else:
+            # Add a space. Its simpler to append a new span element than
+            # inserting text after the last node ends since neither .text
+            # and .tail do the right thing for us.
+            header_content.append(lxml.html.fromstring("<span> </span>"))
+        span_elem = lxml.html.fromstring(
+            f'<span class="spoiler-title" title="{spoiler_title}">({spoiler_title})</span')
+        header_content.append(span_elem)
+        header.drop_tag()
+        spoiler_content.drop_tree()
+    content = lxml.html.tostring(fragment).decode("utf-8")
+    return content
+
+def fix_spoilers_in_text(content: str, language: str) -> str:
+    with override_language(language):
+        spoiler_title: str = _("Open Zulip to see the spoiler content")
+    lines = content.split('\n')
+    output = []
+    open_fence = None
+    for line in lines:
+        m = FENCE_RE.match(line)
+        if m:
+            fence = m.group('fence')
+            lang = m.group('lang')
+            if lang == 'spoiler':
+                open_fence = fence
+                output.append(line)
```

Comment:
We're calling `output.append(line)` thrice which is confusing. Some simplification of the if-else structure here should allow us to make it easier to read and maybe remove the extra calls altogether.